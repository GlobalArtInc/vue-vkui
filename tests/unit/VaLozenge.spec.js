import { mount } from '@vue/test-utils'
import VaLozenge from '@/Lozenge/VaLozenge.vue'

describe('VaLozenge.vue', () => {
  it('renders correctly', () => {
    const wrapper = mount(VaLozenge)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders a span', () => {
    const wrapper = mount(VaLozenge)
    expect(wrapper.contains('span')).toBe(true)
  })

  it('accepts "primary" as prop "type"', () => {
    const wrapper = mount(VaLozenge, {
      propsData: {
        type: 'primary'
      }
    })
    expect(wrapper.props().type).toBe('primary')
  })

  it('prop type of primary results in va-lozenge-primary class', () => {
    const wrapper = mount(VaLozenge, {
      propsData: {
        type: 'primary'
      }
    })
    expect(wrapper.classes()).toContain('va-lozenge-primary')
  })

  it('values passed to slots should render correctly', () => {
    const wrapper = mount(VaLozenge, {
      slots: {
        default: '<span>Hello</span>'
      }
    })
    expect(wrapper.find('span').text()).toBe('Hello')
  })
})
