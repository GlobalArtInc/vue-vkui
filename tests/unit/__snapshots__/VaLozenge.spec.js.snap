// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaLozenge.vue renders correctly 1`] = `
<span
  class="va-lozenge va-lozenge-default"
/>
`;
